diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7cceb2c..630f262 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,13 +59,13 @@ write_basic_package_version_file(
     COMPATIBILITY AnyNewerVersion)
 
 install(FILES
-            "netutilsConfig.cmake"
+            "certifyConfig.cmake"
             "${CMAKE_BINARY_DIR}/certifyConfigVersion.cmake"
         DESTINATION lib/cmake/certify)
 
 install(DIRECTORY ${CMAKE_SOURCE_DIR}/include/
         DESTINATION include
-        FILES_MATCHING PATTERN "*.hpp")
+        FILES_MATCHING PATTERN "*.hpp" PATTERN "*.ipp")
 
 install(TARGETS core
         EXPORT certifyTargets
diff --git a/certifyConfig.cmake b/certifyConfig.cmake
index 272dd90..87313e8 100644
--- a/certifyConfig.cmake
+++ b/certifyConfig.cmake
@@ -1,6 +1,7 @@
 include(CMakeFindDependencyMacro)
 
-find_dependency(Boost COMPONENTS system)
+find_dependency(Boost COMPONENTS system filesystem date_time)
+find_dependency(OpenSSL)
 find_dependency(Threads)
 
-include("${CMAKE_CURRENT_LIST_DIR}/certify-Targets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/certifyTargets.cmake")
diff --git a/include/boost/certify/crlset_parser.hpp b/include/boost/certify/crlset_parser.hpp
index 7174944..29ab461 100644
--- a/include/boost/certify/crlset_parser.hpp
+++ b/include/boost/certify/crlset_parser.hpp
@@ -4,6 +4,7 @@
 #include <boost/asio/buffer.hpp>
 #include <boost/endian/arithmetic.hpp>
 #include <boost/system/system_error.hpp>
+#include <cstdint>
 
 namespace boost
 {
diff --git a/include/boost/certify/detail/keystore_windows.ipp b/include/boost/certify/detail/keystore_windows.ipp
index efcc697..625ef00 100644
--- a/include/boost/certify/detail/keystore_windows.ipp
+++ b/include/boost/certify/detail/keystore_windows.ipp
@@ -6,6 +6,7 @@
 
 #include <sstream>
 #include <wincrypt.h>
+#include <cstdint>
 
 namespace boost
 {
diff --git a/include/boost/certify/detail/spki_blacklist.hpp b/include/boost/certify/detail/spki_blacklist.hpp
index 7833d80..2f456e9 100644
--- a/include/boost/certify/detail/spki_blacklist.hpp
+++ b/include/boost/certify/detail/spki_blacklist.hpp
@@ -2,6 +2,7 @@
 #define BOOST_CERTIFY_DETAIL_SPKI_BLACKLIST_HPP
 
 #include <array>
+#include <cstdint>
 
 namespace boost
 {
diff --git a/include/boost/certify/detail/spki_digest.hpp b/include/boost/certify/detail/spki_digest.hpp
index d9e4ba9..5f937c2 100644
--- a/include/boost/certify/detail/spki_digest.hpp
+++ b/include/boost/certify/detail/spki_digest.hpp
@@ -5,6 +5,7 @@
 #include <boost/make_unique.hpp>
 #include <cassert>
 #include <openssl/x509.h>
+#include <cstdint>
 
 namespace boost
 {
diff --git a/include/boost/certify/impl/crlset_parser.ipp b/include/boost/certify/impl/crlset_parser.ipp
index d41fb7f..853894e 100644
--- a/include/boost/certify/impl/crlset_parser.ipp
+++ b/include/boost/certify/impl/crlset_parser.ipp
@@ -2,6 +2,7 @@
 #define BOOST_CERTIFY_IMPL_CRLSET_PARSER_IPP
 
 #include <boost/certify/crlset_parser.hpp>
+#include <cstdint>
 
 namespace boost
 {
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index c1cda26..03be79c 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -11,5 +11,9 @@ function (certify_verify_add_test test_file)
              COMMAND ${target_name})
 endfunction(certify_verify_add_test)
 
+certify_verify_add_test(extensions.cpp)
 certify_verify_add_test(https_verification_success.cpp)
 certify_verify_add_test(https_verification_fail.cpp)
+certify_verify_add_test(crl_set_parser.cpp)
+certify_verify_add_test(detail_spki_digest.cpp)
+certify_verify_add_test(status_cache.cpp)
\ No newline at end of file
diff --git a/tests/crl_set_parser.cpp b/tests/crl_set_parser.cpp
index 4e5b221..8f29a27 100644
--- a/tests/crl_set_parser.cpp
+++ b/tests/crl_set_parser.cpp
@@ -5,6 +5,7 @@
 #include <iomanip>
 #include <iterator>
 #include <numeric>
+#include <cstdint>
 
 const std::uint8_t array[46] = {
   0x02, 0x00, 0x7b, 0x7d, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
